Automattic/wp-calypso
# Receipts: show more details about Jetpack Search tier

Related to https://github.com/Automattic/wp-calypso/issues/55832 and pbNhbs-ccp-p2

## Proposed Changes

* Shows "Purchase for %(quantity)d search records/requests" instead of "Purchase of %(quantity)d item" for Jetpack Search products
* Adds price breakdown for the Jetpack Search plans with `additional_10k_queries_and_records` pricing tier - this requires PR 169346-ghe-Automattic/wpcom


| Before | After |
| ------------- | ------------- |
| ![image](https://github.com/user-attachments/assets/737fc8b8-6f67-4802-b34d-064f818795e8)  |  ![image](https://github.com/user-attachments/assets/38182498-82d0-4c96-88b3-087251271d77) |

## Why are these changes being made?

* Makes Jetpack Search pricing clearer

## Testing Instructions

This PR requires change 169346-ghe-Automattic/wpcom, apply it first and sandbox public-api.

* Visit http://calypso.localhost:3000/me/purchases/billing
* Find receipts for Jetpack Search, and click "View receipt"

Transactions are cached in `redux-state-<yourUserId>:billingTransactions`, but visiting the billing list first should fetch the fresh data with pricing tier.

## Pre-merge Checklist
